test/e2e/certmanager/certmanager_test.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certmanager
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/kueue/apis/config/v1beta1"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	workloadappwrapper "sigs.k8s.io/kueue/pkg/controller/jobs/appwrapper"
+	workloadjob "sigs.k8s.io/kueue/pkg/controller/jobs/job"
+	workloadjobset "sigs.k8s.io/kueue/pkg/controller/jobs/jobset"
+	workloadlws "sigs.k8s.io/kueue/pkg/controller/jobs/leaderworkerset"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	awtesting "sigs.k8s.io/kueue/pkg/util/testingjobs/appwrapper"
+	testingjob "sigs.k8s.io/kueue/pkg/util/testingjobs/job"
+	testingjobset "sigs.k8s.io/kueue/pkg/util/testingjobs/jobset"
+	testinglws "sigs.k8s.io/kueue/pkg/util/testingjobs/leaderworkerset"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("CertManager", ginkgo.Ordered, func() {
+	var (
+		ns           *corev1.Namespace
+		defaultRf    *kueue.ResourceFlavor
+		localQueue   *kueue.LocalQueue
+		clusterQueue *kueue.ClusterQueue
+	)
+
+	ginkgo.BeforeAll(func() {
+		configurationUpdate := time.Now()
+		config := defaultKueueCfg.DeepCopy()
+		config.InternalCertManagement = &v1beta1.InternalCertManagement{
+			Enable: ptr.To(false),
+		}
+		util.ApplyKueueConfiguration(ctx, k8sClient, config)
+		util.RestartKueueController(ctx, k8sClient)
+		ginkgo.By("Checking Kueue certificates are ready", func() {
+			util.WaitForCertificateReady(ctx, k8sClient,
+				"serving-cert", "kueue-system", 5*time.Minute)
+		})
+		ginkgo.By("Checking webhook CA bundles are injected", func() {
+			util.WaitForWebhookCABundleReady(ctx, k8sClient, 2*time.Minute)
+		})
```

Comment:
What is the purpose of these checks? Are they needed to prevent flakes, or just debuggability?

I think they should not be needed, because Kueue would not be "Available" until the webhooks endpoint is responding, because of our readiness probe: https://github.com/kubernetes-sigs/kueue/blob/69a9538f234aaeed55d825ac04e8c1338e62c5a7/cmd/kueue/main.go#L393-L399

If these checks are indeed not necessary, I would suggest to start simple and drop them. We can always add debuggability code as needed.

Reply:
Yeah, you are right. This code was mostly for debugging. Let me go ahead and remove that code.